src/main/java/mate/controller/DeleteCarController.java

```diff
@@ -0,0 +1,34 @@
+package mate.controller;
+
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.service.CarService;
+
+@WebServlet(urlPatterns = "/cars/delete")
+public class DeleteCarController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate");
+    private final CarService carService = (CarService) injector.getInstance(CarService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        Long id = Long.valueOf(request.getParameter("id"));
+        carService.delete(id);
+        response.sendRedirect(request.getContextPath() + "/cars/delete");
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Car> currentCars = carService.getAll();
+        req.setAttribute("cars", currentCars);
+        req.getRequestDispatcher("/WEB-INF/views/cars/delete.jsp").forward(req, resp);
+    }
```

Comment:
In **all** delete operations/controllers I don't like an idea that they are combined with get controllers.
Please split this logic to corresponding controllers.
GetAll...Controller will have doGet method and provide user list of some objects, Delete...Controller will have doPostMethod, perform deletion logic and will redirect user to respective GetAll page

Because now it's really strange that after object creation we redirect user to delete page where he gets list of objects